core/libp2p/multi/multiaddress/multiaddress.cpp

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <algorithm>
+#include <numeric>
+#include <stdexcept>
+
+#include "libp2p/multi/multiaddress.hpp"
+extern "C" {
+#include "libp2p/multi/multiaddress/c-utils/protoutils.h"
+}
+
+namespace {
+  // string representations of protocols
+  constexpr std::string_view kIp4 = "/ip4";
```

Comment:
why no namespace?

Reply:
I want those functions and constants to be in an anonymous namespace